components/future_ext/src/cancel.rs

```diff
@@ -36,6 +47,9 @@ where
     F: Future + Send + 'static,
     F::Output: Send,
 {
+    /// Token for metrics
+    token: String,
```

Comment:
Maybe we can declare the token as a generic parameter T, where `T: AsRef<str> + 'static`. And it can avoid memory allocation for some cases where only a static constant string is used.